elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryConstructor.java

```diff
@@ -0,0 +1,507 @@
+        return sortClauses.keySet().stream()
+                .map(this::expandJoinToPath)
+                .map(this::extractPathElements)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+    }
+
+    /**
+     * Given the set of group by dimensions, extract any entity relationship traversals that require joins.
+     * This method takes in a {@link SQLAnalyticView} because the sql join path meta data is stored in it.
+     *
+     * @param groupByDimensions The list of dimensions we are grouping on.
+     * @param queriedTable queried analytic view
+     * @return A set of path elements that capture a relationship traversal.
+     */
+    private Set<Path.PathElement> extractPathElements(Set<ColumnProjection> groupByDimensions,
+                                                      SQLAnalyticView queriedTable) {
+        return resolveSQLColumns(groupByDimensions, queriedTable).stream()
+                .filter((dim) -> dim.getJoinPath() != null)
+                .map(SQLColumn::getJoinPath)
+                .map(this::extractPathElements)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Given a path , extracts any entity relationship traversals that require joins.
+     *
+     * @param path The path
+     * @return A set of path elements that capture a relationship traversal.
+     */
+    private Set<Path.PathElement> extractPathElements(Path path) {
+        return path.getPathElements().stream()
+                .filter((p) -> dictionary.isRelation(p.getType(), p.getFieldName()))
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+    }
+
+    /**
+     * Translates a filter expression into SQL.
+     *
+     * @param expression The filter expression
+     * @param columnGenerator A function which generates a column reference in SQL from a FilterPredicate.
+     * @return A SQL expression
+     */
+    private String translateFilterExpression(FilterExpression expression,
+                                             Function<FilterPredicate, String> columnGenerator) {
+        FilterTranslator filterVisitor = new FilterTranslator();
+
+        return filterVisitor.apply(expression, columnGenerator);
+    }
+
+    /**
+     * Converts a filter predicate into a SQL WHERE/HAVING clause column reference.
+     *
+     * @param predicate The predicate to convert
+     * @return A SQL fragment that references a database column
+     */
+    private String generateColumnReference(FilterPredicate predicate) {
+        return generateColumnReference(predicate.getPath());
+    }
+
+    /**
+     * Converts a filter predicate path into a SQL WHERE/HAVING clause column reference.
+     *
+     * @param path The predicate path to convert
+     * @return A SQL fragment that references a database column
+     */
+    private String generateColumnReference(Path path) {
```

Comment:
In general, I would have preferred that we kept the old function names and helper functions untouched where possible.  That code was already inspected.

Reply:
Can we revert to the old structure of helpers?

Reply:
The structure is not changed, the old method is name `generateWhereClauseColumnReference()`, but it is not only used in where clause.